metagraph/src/common/unix_tools.cpp

```diff
@@ -111,6 +115,30 @@ size_t get_curr_RSS()
 }
 
 
+size_t get_max_files_open() {
+    struct rlimit lim;
+    getrlimit(RLIMIT_NOFILE, &lim);
+    rlim_t max_files = lim.rlim_cur;
+    return max_files;
+}
+
+int get_num_fds() {
+    FILE *p = popen(fmt::format("lsof -p {} | tail -n +2 | wc -l", getpid()).c_str(), "r");
```

Comment:
Isn't this unsafe since it's getting this value from an external process?

Maybe add a TODO to get this value from `lsof.h` or some POSIX API calls

Reply:
Ah, good that you found that. When I quickly checked it on the internet, I couldn't find a way to get around invoking `lsof` in a separate process.

Reply:
I checked it out but it doesn't look straightforward. Not sure that makes sense.

Reply:
ok, then let's add a TODO comment here so we can find it later

Reply:
Not sure that would be a good thing to spend time on... The current approach should work fine.